dvc/cli/__init__.py

```diff
@@ -48,6 +49,11 @@ def main(argv=None):  # noqa: C901
     from dvc.exceptions import DvcException, NotDvcRepoError
     from dvc.logger import FOOTER, disable_other_loggers
 
+    if sys.stderr.closed:  # pylint: disable=using-constant-test
+        logging.disable()
+    elif sys.stdout.closed:  # pylint: disable=using-constant-test
+        logging.disable(logging.INFO)
```

Comment:
in `dvc.daemon` we double fork and close the standard file descriptors (to avoid duplicating the parent's FDs in the child daemon). On Linux, we never actually call `exec` after the double fork, we just directly call `cli.main`, which results in stdout and stderr still being closed in the daemon process. Making any logging calls in this state results in an exception due to writing to a closed file descriptor.

In normal use this leads to a silent failure when we run the DVC analytics or updater daemon in verbose/debug mode, but in pyinstaller builds this causes pyinstaller to generate an unhandled exception error message that gets printed to the original parent terminal.

Reply:
@pmrowla, can you please add this as a comment in the code itself? Thanks.

Reply:
Why this didn't happen before?

Reply:
@daavoo I think it probably did and we just never noticed, because we don't run any thorough tests with the pyinstaller builds.

In pip-installed DVC, we won't ever see this exception because it just fails silently (since there is no stderr/stdout to log the exception to, the python interpreter will just exit). It gets caught and logged in the pyinstaller builds since pyinstaller's separate bootloader is still running separately from the pyinstaller package's python interpreter

Reply:
Even if we did have more end-to-end tests, catching this would still be difficult because it requires watching the system for spawned python (or pyinstaller) processes, figuring out which one is the daemonized child process, and checking exit codes when that process terminates.